Sources/CodegenCLI/Commands/GenerateOperationManifest.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import ArgumentParser
+import ApolloCodegenLib
+
+public struct GenerateOperationManifest: ParsableCommand {
+  
+  // MARK: - Configuration
+  
+  public static var configuration = CommandConfiguration(
+    abstract: "Generate Persisted Queries operation manifest based on a code generation configuration."
+  )
+  
+  @OptionGroup var inputs: InputOptions
+  
+  @Flag(
+    name: .long,
+    help: "Ignore Apollo version mismatch errors. Warning: This may lead to incompatible generated objects."
+  )
+  var ignoreVersionMismatch: Bool = false
+  
+  // MARK: - Implementation
+  
+  public init() { }
+  
+  public func run() throws {
+    try _run()
+  }
+  
+  func _run(
+    fileManager: FileManager = .default,
+    codegenProvider: CodegenProvider.Type = ApolloCodegen.self,
+    logger: LogLevelSetter.Type = CodegenLogger.self
+  ) throws {
+    logger.SetLoggingLevel(verbose: inputs.verbose)
+    
+    try checkForCLIVersionMismatch(
+      with: inputs,
+      ignoreVersionMismatch: ignoreVersionMismatch
+    )
+    
+    var configData: Data?
+    switch (inputs.string, inputs.path) {
+    case let (.some(string), _):
+      configData = try string.asData()
+    case let (nil, path):
+      configData = try fileManager.unwrappedContents(atPath: path)
+    }
+    
+    guard let data = configData else {
```

Comment:
It seems like the only reason we need the `guard` here is because of `configData` being defined as optional when both `asData` and `unwrappedContents(atPath:)` return a non-optional type. I'm not sure there is any value in the error message with those return types and command validation should ensure that one of `--path` or `--string` must be supplied with the command.